scripts/generator.ts

```diff
@@ -364,9 +364,34 @@ const elementFilesMap = await prepareElementFiles(descriptions);
 
 const sourceFiles = Array.from(elementFilesMap.entries(), ([element, data]) => generateReactComponent(element, data));
 
+function generateIndexDtsFile(elementNames: readonly string[]): SourceFile {
+  const sourceLines = [
+    `// Workaround for VSCode, enables TypeScript auto-imports form package.json`,
+    ...elementNames.map((elementName) => `import './${elementName}.js';`),
+  ];
+
+  return ts.createSourceFile(
+    resolve(rootDir, 'index.d.ts'),
+    sourceLines.join('\n'),
+    ts.ScriptTarget.ES2019,
+    undefined,
+    ts.ScriptKind.TS,
+  );
+}
+
+const indexDtsFile = generateIndexDtsFile(sourceFiles.map(({ fileName }) => basename(fileName, '.ts')));
+
+const indexJsFile = ts.createSourceFile(
+  resolve(rootDir, 'index.js'),
```

Comment:
Do we really need `index.js`? Wouldn't it work with just `index.d.ts`?

Reply:
Well, no, but `index.d.ts` presence makes `import '@hilla/react-components';` work for TypeScript applications, there is no error in the IDE. Removing `index.js` would break such imports in “runtime” or whenever frontend build attempts to resolve them.

I thought that “runtime” errors without an IDE highlight are unexpected, so added the `index.js` to prevent it.

Reply:
Well, it still looks somewhere wrong but ok, let's do it this way